src/main/java/com/google/sps/utils/data/DataCollection.java

```diff
@@ -25,139 +25,163 @@
-      return species;
+            System.out.printf("%-35s %-30s %-25s %-10s %-15s %n", commonName, binomialName, population, status, trend);
+            //   Species species = new Species(commonName, binomialName, status, trend, population, notes, imageLink, url);
+            Species species = new Species.Builder()
+                                        .withCommonName(commonName)
+                                        .withBinomialName(binomialName)
+                                        .withStatus(status)
+                                        .withPopulationTrend(trend)
+                                        .withPopulation(population)
+                                        .withWikipediaNotes(notes)
+                                        .withImageLink(imageLink)
+                                        .withCitationLink(url)
+                                        .build();
+            return species;
+        }
+        return null;
     }
-    return null;
-  }
 
-  /**
-   * Checks if the species is already in Datastore
-   * @return true if already stored, false otherwise
-   */
-  private static boolean speciesAlreadyStored(String binomialName) {
-    Key key = keyFactory.newKey(binomialName);
-    Entity speciesEntity = datastore.get(key);
-    return speciesEntity != null;
-  }
+    /**
+    * Checks if the species is already in Datastore
+    * @return true if already stored, false otherwise
+    */
+    private static boolean speciesAlreadyStored(String binomialName) {
+        Key key = keyFactory.newKey(binomialName);
+        Entity speciesEntity = datastore.get(key);
+        return speciesEntity != null;
+    }
 
-  /**
-   * Retrieves apiMap for the passed species and adds additional info to species
-   * @param species: non-null species that isn't already stored in Datastore
-   */
-  public static void addApiInfo(Species species) {
-    // Add API-side fields if available
-    try {
-      String apiJSON = SpeciesAPIRetrieval.getJSON(API_URL, species.getBinomialName());
-      Map apiMap = SpeciesAPIRetrieval.convertJSONToMap(apiJSON);
-      SpeciesAPIRetrieval.addAPISpeciesInfo(species, apiMap);
-    } catch (Exception e) {
-      System.err.println("Exception occurred retrieving API data: " + e);
-      e.printStackTrace();
+    /**
+    * Retrieves apiMap for the passed species and adds additional info to species
+    * @param species: non-null species that isn't already stored in Datastore
+    */
+    public static void addApiInfo(Species species) {
+        // Add API-side fields if available
+        try {
+            String apiDataJSON = SpeciesAPIRetrieval.getJSON(DATA_URL, species.getBinomialName());
+            Map apiDataMap = SpeciesAPIRetrieval.convertJSONToMap(apiDataJSON);
+            SpeciesAPIRetrieval.addAPISpeciesInfo(species, apiDataMap);
+        } catch (Exception e) {
+            System.err.println("Exception occurred retrieving API data: " + e);
```

Comment:
Can you add the exception string as a variable so that it's not a magic string? It's used in `addGeoInfo()` too, so it might clean it up a bit

Reply:
What's a magic string? If I understand correctly, do you mean put "Exception occurred retrieving API data: " into a separate variable?